cmake finds a DTC from Zephyr-SDK version, it tries to execute it, and it fails

**Describe the bug**
A clear and concise description of what the bug is.
What have you tried to diagnose or workaround this issue?
this line https://github.com/zephyrproject-rtos/zephyr/blob/603c24dec913749c0198b2a10afc52ac5c3443db/cmake/host-tools.cmake#L50
figured out that my dtc is supposed to be `/usr/zephyr-sdk-0.11.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc` instead of `/usr/bin/dtc` which shouldn't work anyway because I have DTC 1.4.5.
But this specific version of dtc that I get from the SDK automatically fails and gives no stdout but stderr:
`/usr/zephyr-sdk-0.11.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc: error while loading shared libraries: libunwind.so.8: cannot open shared object file: No such file or directory`
(I do have libunwind installed)
because of this line https://github.com/zephyrproject-rtos/zephyr/blob/603c24dec913749c0198b2a10afc52ac5c3443db/cmake/host-tools.cmake#L63
since it has no output cmake says
```
CMake Error at /content/zephyrproject/zephyr/cmake/host-tools.cmake:65 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  /content/zephyrproject/zephyr/cmake/app/boilerplate.cmake:490 (include)
  /content/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /content/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


CMake Error at /content/zephyrproject/zephyr/cmake/host-tools.cmake:70 (if):
  if given arguments:

    "VERSION_GREATER" "1.4.6"

  Unknown arguments specified
Call Stack (most recent call first):
  /content/zephyrproject/zephyr/cmake/app/boilerplate.cmake:490 (include)
  /content/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /content/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:5 (find_package)
```


**To Reproduce**
Steps to reproduce the behavior:
1. install sdk
2. west build -b qemu_x86 samples/hello_world
3. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

CMake should make a sanity check to see if ${dtc_version_output} is empty before making the regex check, this will prevent this bug.
The DTC of Zephyr-SDK shouldn't have libunwind issues, either.

--- a/cmake/host-tools.cmake
+++ b/cmake/host-tools.cmake
@@ -59,17 +59,26 @@ if(DTC)
     COMMAND
     ${DTC} --version
     OUTPUT_VARIABLE dtc_version_output
+    ERROR_VARIABLE  dtc_error_output
+    RESULT_VARIABLE dtc_status
     )
-  string(REGEX MATCH "Version: DTC ([0-9]+[.][0-9]+[.][0-9]+).*" out_var ${dtc_version_output})
 
-  # Since it is optional, an outdated version is not an error. If an
-  # outdated version is discovered, print a warning and proceed as if
-  # DTC were not installed.
-  if(${CMAKE_MATCH_1} VERSION_GREATER ${MIN_DTC_VERSION})
-    message(STATUS "Found dtc: ${DTC} (found suitable version \"${CMAKE_MATCH_1}\", minimum required is \"${MIN_DTC_VERSION}\")")
+  if(${dtc_status} EQUAL 0)
+    string(REGEX MATCH "Version: DTC ([0-9]+[.][0-9]+[.][0-9]+).*" out_var ${dtc_version_output})
+
+    # Since it is optional, an outdated version is not an error. If an
+    # outdated version is discovered, print a warning and proceed as if
+    # DTC were not installed.
+    if(${CMAKE_MATCH_1} VERSION_GREATER ${MIN_DTC_VERSION})
+      message(STATUS "Found dtc: ${DTC} (found suitable version \"${CMAKE_MATCH_1}\", minimum required is \"${MIN_DTC_VERSION}\")")
+    else()
+      message(WARNING
+        "Could NOT find dtc: Found unsuitable version \"${CMAKE_MATCH_1}\", but required is at least \"${MIN_DTC_VERSION}\" (found ${DTC}). Optional devicetree error checking with dtc will not be performed.")
+      set(DTC DTC-NOTFOUND)
+    endif()
   else()
     message(WARNING
-      "Could NOT find dtc: Found unsuitable version \"${CMAKE_MATCH_1}\", but required is at least \"${MIN_DTC_VERSION}\" (found ${DTC}). Optional devicetree error checking with dtc will not be performed.")
+      "Could NOT find working dtc: Found dtc (${DTC}), but failed to load with:\n ${dtc_error_output}")
     set(DTC DTC-NOTFOUND)
   endif()
 endif()

